projects/Buttons_and_ADC/src/main.c

```diff
@@ -0,0 +1,68 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+static void prv_button_interrupt_handler(const GpioAddress *adc_address, void *context) {
+  GpioAddress *adc_position = context;
+  uint16_t adc_data = 0;
+  StatusCode adc_converted_data = adc_read_converted_pin(*adc_position, &adc_data);
```

Comment:
This is good. However, I would not name this variable adc_converted_data since this returns a status code and not the actual data so may confuse whoever is looking at your code. Also you don't actually need to declare a separate variable for this. You can toss this directly in the if(...) statement

eg `if(adc_read_converted_pin(*adc_position, &adc_data) == STATUS_CODE_OK)`